tests/test_task.py

```diff
@@ -1238,21 +1242,24 @@ def test_return_value(self):
                     transient_run=True,
                     store_result=True,
                     args=(expected_value,),
-                    task_name=f'first re-run transient')
+                    task_name='first re-run transient')
                 value = value_task.get()
                 self.assertEqual(value, expected_value)
+                task_graph.close()
+                task_graph.join()
             else:
                 with self.assertRaises(RuntimeError):
                     value_task = task_graph.add_task(
                         func=_return_value_once,
                         transient_run=True,
                         store_result=True,
                         args=(expected_value,),
-                        task_name=f'expected error {iteration_id}')
+                        task_name='expected error {iteration_id}')
                     value = value_task.get()
 
-            task_graph.close()
-            task_graph.join()
+                with self.assertRaises(RuntimeError):
+                    task_graph.join()

```

Comment:
We're expecting the `join` to now also propagate and raise the same exception from the above task? Can we avoid a `close` here because the `join` exception will handle that?

Reply:
Oh interesting!  Do you mean in the general case or in this specific test?

In this test, I think you're right, closing the graph isn't strictly necessary.  The `task_graph.join()` will also terminate the graph when the exception is raised.  Because we only have one task to execute, we also happen to know that the whole graph has completed.

In general though, I think the `close` is still useful to have since it's possible to join the graph within the pipeline. 

```text

<define tasks>
graph.join()   <-- might raise exception

<define more tasks>
graph.close()   # No more tasks can be added after this
graph.join()  <-- also might raise exception
```

Do you think that makes sense here?  I'm a little wary of changing taskgraph behavior, so I want to be sure that this is a reasonable idea!

Reply:
Yea this makes sense, thanks. I was mostly curious about this specific case and the decision to drop the `task_graph.close()` call.